es/index.js

```diff
@@ -172,13 +172,13 @@ function fixConfig(axios, config) {
  */
 export default function axiosRetry(axios, defaultOptions) {
   axios.interceptors.request.use(config => {
-    const currentState = getCurrentState(config);
+    const currentState = getCurrentState(axios);
```

Comment:
Does this not use a single state per whole axios instance? Will that not make requesting separate resources with the same axios instance impossible, because their retry states will be the same instead of independent?

Reply:
Indeed. But it does not cause too much trouble I guess... I personally tend to create as many custom Axios instances as microservices (apis) my app talks to, e.g.:

```javascript
const todosApiClient = axios.create({ baseURL: 'https://todos-api.com' });
const todosApiClientWithRetry = axiosRetry(todosApiClient, { ... });

const usersApiClient = axios.create({ baseURL: 'https://users-api.com' });
const usersApiWithLogger = axiosLogger(usersApiClient, { ... });

// etc.
```

Reply:
While that is true, and it is nice that it works for you, it is certainly not standard and very much unexpected behavior for anyone who reuses axios instance for different MS. Or for example has axios instance for a single domain but the domain is API gateway and so different resources can have vastly different availability.

Right now this change would make safe axios instance reuse impossible.

Keeping retry state in `config.adapter` as seen in https://github.com/softonic/axios-retry/pull/76/files would not work?

Reply:
The API gateway is an interesting use case, we should indeed find a way to support it.
On one side, I'm not super keen to follow the `config.adapter` approach as we shouldn't modify what we don't own. On the other side I don't see how we could pass the axios-retry from one retry to the next in a reliable way...

Reply:
...unless we decorate Axios `request` method in order to store the axios-retry state in a local "cache", the cache could be a WeakMap and the key, the config passed to the `request` method.

Reply:
The WeakMap idea sounds pretty good to me, compatibility with ie < 11 aside. Either with the `request` object itself as key, as you say, or on some id that is generated in an interceptor.

Reply:
One solution could come from Axios itself https://github.com/axios/axios/pull/2207